core/templates/core/layout.html

```diff
@@ -0,0 +1,257 @@
+                  href="javascript:void(0)"></a>
+              </span>
+            </li>
+          </ul>
+        </div>
+      </div>
+      <div class="pt-0 navbar-collapse">
+        <ul class="pt-0 navbar-nav ml-auto">
+          <a class="py-2" href="{% url 'core:profile-view' %}">
+            <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" fill="currentColor"
+              class="bi bi-person-circle" viewBox="0 0 16 16">
+              <path style="color: #F33F3F;" d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z" />
+              <path style="color: #F33F3F;" fill-rule="evenodd"
+                d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z" />
+            </svg>
+          </a>
+          <li style="margin-left: 0px;" class="nav-item">
+            <a class="nav-link" id="user-name"></a>
+          </li>
+        </ul>
+      </div>
+    </nav>
+  </header>
+  {% endblock %}
+
+  <!-- Page Content -->
+  <!-- Banner Starts Here -->
+  {% block content %}{% endblock %}
+  <!-- Footer -->
+  {% block footer %}{% endblock %}
+  <!-- Bootstrap core JavaScript -->
+  <script src="{% static 'vendor/jquery/jquery.min.js' %}"></script>
+  <script src="{% static 'vendor/bootstrap/js/bootstrap.bundle.min.js' %}"></script>
+  <script src="https://cdn.jsdelivr.net/npm/sweetalert2@11.4.32/dist/sweetalert2.all.min.js"></script>
+
+
+  <!-- Additional Scripts -->
+  <script src="{% static 'assets/js/custom.js' %}"></script>
+  <script src="{% static 'assets/js/owl.js' %}"></script>
+  <script src="{% static 'assets/js/slick.js' %}"></script>
+  <script src="{% static 'assets/js/isotope.js' %}"></script>
+  <script src="{% static 'assets/js/accordions.js' %}"></script>
+
+  <script>
+    // This function is called to clear session storage when user logout
+    function logout() {
+      sessionStorage.removeItem("access")
+      sessionStorage.removeItem("refresh")
+      sessionStorage.removeItem("user_id")
+      window.location.href = "{% url 'authentication:login-view' %}"
+    }
+
+    function isLogin() {
+      const Bearer_access = sessionStorage.access
+      const Bearer_refresh = sessionStorage.refresh
+      const user_id = sessionStorage.user_id
+      const header = new Headers({
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer ' + Bearer_access
+      });
+      if ((Bearer_access != undefined) & (Bearer_refresh != undefined) & (user_id != undefined)) {
+        token = {
+          "Bearer_access": Bearer_access,
+          "Bearer_refresh": Bearer_refresh,
+          "user_id": user_id,
+          "header": header
+        }
+        return token;
+      }
+      else {
+        return false
+      }
+    }
+
+    // Gettin the name of the user
+    async function getUser(token) {
+
+      const url = "/auth/user/" + token.user_id
```

Comment:
I think we should get url via view name instead of hardcoding it, as done above

Reply:
We can't use view name here as `{% url 'authentication:user-view' pk=user_id %}`. Because we need a `user_id` to complete the url and this function is called after loading the page. We have no `user_id` at the time of rendering.